Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -90,17 +89,95 @@ if(WIN32)
         win32/res/Dirt.ICO
         win32/res/headspin.ico
     )
+elseif(APPLE)
+    set(plClient_SOURCES ${plClient_SOURCES}
+        Mac-Cocoa/main.mm
+        Mac-Cocoa/PLSKeyboardEventMonitor.mm
+        Mac-Cocoa/PLSView.mm
+        Mac-Cocoa/PLSLoginWindowController.mm
+        Mac-Cocoa/PLSPatcherWindowController.mm
+        Mac-Cocoa/PLSPatcher.mm
+        Mac-Cocoa/PLSServerStatus.mm
+        Mac-Cocoa/StringTheory_NSString.mm
+    )
+    set(plClient_HEADERS ${plClient_HEADERS}
+        Mac-Cocoa/PLSKeyboardEventMonitor.h
+        Mac-Cocoa/PLSView.h
+        Mac-Cocoa/PLSLoginWindowController.h
+        Mac-Cocoa/PLSPatcherWindowController.h
+        Mac-Cocoa/PLSPatcher.h
+        Mac-Cocoa/PLSServerStatus.h
+        Mac-Cocoa/StringTheory_NSString.h
+    )
+    set(RESOURCES 
+        Mac-Cocoa/MainMenu.xib
+        Mac-Cocoa/PLSLoginWindowController.xib
+        Mac-Cocoa/PLSPatcherWindowController.xib
+        Mac-Cocoa/banner@2x.jpeg
+        win32/res/Dirt.ICO
+    )
+    
+    list(APPEND plClient_RESOURCES
+        win32/res/resource.h
+        win32/res/banner.bmp
+        win32/res/Dirt.ICO
+        win32/res/headspin.ico
+    )
 else()
     list(APPEND plClient_SOURCES
         main.cpp
     )
 endif()
 
-plasma_executable(plClient WIN32 CLIENT INSTALL_PDB
-    SOURCES
-        ${plClient_SOURCES} ${plClient_HEADERS}
-        ${plClient_TEXT} ${plClient_RESOURCES}
-)
+if(APPLE)
+    plasma_executable(plClient CLIENT INSTALL_PDB SOURCES ${plClient_SOURCES} ${plClient_HEADERS} ${RESOURCES})
+    set(MAIN_NIBFILE "MainMenu")
+    set_target_properties(plClient PROPERTIES
+        BUNDLE True
+        MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/Mac-Cocoa/Info.plist.in"
+        MACOSX_BUNDLE_GUI_IDENTIFIER org.openuru.plClient
+        MACOSX_BUNDLE_BUNDLE_VERSION "0.1"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "0.1"
```

Comment:
This probably has to be injected through CMake. Is that possible?

Reply:
It should be...

https://github.com/H-uru/Plasma/blob/master/cmake/hsBuildInfo.inl.cmake
https://github.com/H-uru/Plasma/blob/faf77219262de199914a17e6ffbf2052ae75bd85/CMakeLists.txt#L24-L30

Reply:
I believe MACOSX_BUNDLE_BUNDLE_VERSION is suppose to be the build number. MACOSX_BUNDLE_SHORT_VERSION_STRING is the public/human readable version. Is that assigned anywhere by CMake?

Reply:
Currently we don't have anything that represents an actual "build number". The OpenUru client does, which is set from the Jenkins run number.

We could set it to the GitHub run number, which is probably fine since it just needs to increment and doesn't need to be sequential

Reply:
What does PRODUCT_BUILD_ID represent?

Reply:
Once upon a time, this was the sort of build number that you're looking for... however, it must match what the server is expecting otherwise the client gets rejected as being "too old" or "too new", so it's been `918` since 2011 and will forever be so 🙃